python/pyarrow/tests/test_parquet.py

```diff
@@ -579,6 +579,22 @@ def test_pandas_parquet_native_file_roundtrip(tempdir, use_legacy_dataset):
     tm.assert_frame_equal(df, df_read)
 
 
+@parametrize_legacy_dataset
+def test_parquet_read_from_buffer(tempdir, use_legacy_dataset):
```

Comment:
this test is alraedy passing on master, but just noticed that we are not explicitly testing this use case (only with a `BytesIO` or `BufferReader` objects)